packages/flame/lib/src/components/timer_component.dart

```diff
@@ -32,6 +38,17 @@ class TimerComponent extends Component {
     );
   }
 
+  @override
+  @mustCallSuper
+  FutureOr<void> onLoad() {
+
+    if (tickOnLoad) {
+      onTick();
+    }
+
+    return super.onLoad();
```

Comment:
It should load before it ticks, if there would be something in super in the future.